### PR TITLE
build: shell out to python to check host pip rather than environment's

### PR DESCRIPTION
**Pull request by cjp256**
Importing pip directly is checking the environment in which charmcraft
is running, not the version of pip that will be used by host's pip3.

Without changing the check behavior, simply call out to python3 and
perform the check.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

### CODE REVIEW

**Comment by cjp256** [comment]
This is in #316's queue as a requirement to pass integration tests on Bionic with classic snap.

**Review by sergiusens — APPROVED** [review]
Looks sane, we cannot rely on the pip bundled with charmcraft as it will be tied to the interpreter charmcraft is using if directly imported.

File: charmcraft/commands/build.py [hunk 1]
```diff
@@ -55,13 +55,16 @@
...
-        return False
+    cmd = [
+        "python3",
+        "-c",
+        (
+            "from pip.commands.install import InstallCommand; "
+            'assert InstallCommand().cmd_opts.get_option("--system") is not None'
+        ),
+    ]
+    proc = subprocess.run(cmd, capture_output=True)
```
Comment (facundobatista) [inline charmcraft/commands/build.py:66]: `capture_output` does not exist in Python 3.6. Why we need it anyway?

**Review by facundobatista — CHANGES_REQUESTED** [review]

File: charmcraft/commands/build.py (hunk 1 above)
Comment (sergiusens) [reply charmcraft/commands/build.py:66]: For context, subprocess.run is processed using charmcraft's python interpreter (3.8), not the one on the system, the python in the arguments to subprocess.run is the one on the system

Comment (facundobatista) [reply charmcraft/commands/build.py:66]: Note that charmcraft may still be installed in Bionic from PyPI.

Comment (sergiusens) [reply charmcraft/commands/build.py:66]: I thought we agreed to baseline on 3.8, was that not the case? python3.8 is available on 18.04.

Comment (facundobatista) [reply charmcraft/commands/build.py]: Let's talk about this. It's not clear to me where/how we want charmcraft to run.

**Review by facundobatista — APPROVED** [review]
Wonderful, thanks!

**Pull request merged**